src/pudl/workspace/setup.py

```diff
@@ -119,6 +119,33 @@ def derive_paths(pudl_in, pudl_out):
 
     ferc1_db_file = pathlib.Path(pudl_settings["sqlite_dir"], "ferc1_xbrl.sqlite")
     pudl_settings["ferc1_xbrl_db"] = "sqlite:///" + str(ferc1_db_file.resolve())
+    pudl_settings["ferc1_xbrl_descriptor"] = pathlib.Path(
```

Comment:
Maybe you already saw, but it seems like it's now possible to specify a database table as the location of the resource specified in a tabular datapackage, if you wanted to hook the resource descriptor directly to the SQLite DB so that they can be used as a unit by the Frictionless Framework:

https://framework.frictionlessdata.io/docs/formats/sql.html#sqlite

Reply:
Also omfg it is so far past time to replace `pudl_settings` with a real Pydantic settings model.

Reply:
I have read a bit on that, but don't quite get how it works in practice. I have included a path to the database, and a [sql dialect](https://v4.framework.frictionlessdata.io/docs/references/api-reference#sqldialect) that specifies the table for each resource, but I don't really get how they're used beyond that.

Reply:
Also strong agree on making `pudl_settings` a Pydantic class. It takes some digging to figure out what's included there